src/Symfony/Component/TypeInfo/Type/EnumType.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\TypeInfo\Type;
+
+use Symfony\Component\TypeInfo\Exception\InvalidArgumentException;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ * @author Baptiste Leduc <baptiste.leduc@gmail.com>
+ */
+class EnumType extends ObjectType
```

Comment:
Question: Maybe I'm mistaken but have you forgotten the final keyword here ?
```suggestion
final class EnumType extends ObjectType
```

Reply:
Since `BackedEnumType` extends `EnumType` we cannot make it final :wink:

Reply:
But you can add `@final`, right?

Reply:
Since the class is already extended, won't it make the final keyword kinda obsolete here ?

Reply:
Even if both classes are in the same vendor,`DebugClassLoader` would trigger a deprecation notice so we cannot make it `@final`